src/adv.py

```diff
@@ -33,19 +44,84 @@
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
 
+
+directions = ['n', 'e', 's', 'w']
+actions = ['t', 'd', 'take', 'drop']
+
 #
 # Main
 #
+playername = input('Sorcerer: What be thy name? ')
+player = Player(playername,room['outside'])
+print(f"Sorceror: May you find your adventure worthy, {player.name}")
+helpstring = print("You can use the following commands: \n N: Move North \n E: Move East \n S: Move South \n W: Move West \n T or Take + Item: Take item \n D or Drop + Item: Drop Item \n I: examine Inventory \n H: Help (prints the commands) \n Q: Quit if this be thy wish")
 
 # Make a new player object that is currently in the 'outside' room.
 
+
 # Write a loop that:
 #
+while True:
+    if player.room.items != []:
+        for item in player.room.items:
+            print(item)
+    else:
+        print('No items in this room.')
+    print(
+        "Your current room:",
+        "\n",
+        player.room.name,
+         "\n", 
+         player.room.description
+         )
+    direction = (input("Input a command: ")).lower().split()
+    if len(direction) == 1:
+        if direction[0] == "q": 
+            break
+        elif direction[0] == "n":
+            if hasattr(player.room, "n_to"):
+                player.room = player.room.n_to
+            else: 
+                print("You don't find a room to the north of you")
+        elif direction[0] == "e":
+            if hasattr(player.room, "e_to"):
+                player.room = player.room.e_to
+            else:
+                print("You don't find a room to the east of you")
+        elif direction[0] == "s":
+            if hasattr(player.room, "s_to"):
+                player.room = player.room.s_to
+            else:
+                print("You don't find a room to the south of you")
+        elif direction[0] == "w":
+            if hasattr(player.room, "w_to"):
+                player.room = player.room.w_to
+            else:
+                print("You don't find a room to the west of you")
```

Comment:
Try to factor out the moving logic to the Player class and only call that method inside the adv.py file.
`getattr` will be useful since you are already using `hasattr`